.github/workflows/release.yml

```
@@ -0,0 +1,80 @@
+name: Release
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  release:
+    name: Release
+    permissions:
+      actions: write
+      checks: write
+      contents: write
+      deployments: write
+      discussions: write
+      id-token: write
+      issues: write
+      packages: write
+      pages: write
+      pull-requests: write
+      repository-projects: write
+      security-events: write
+      statuses: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set Env Variables
+        run: |
+          current_branch=${GITHUB_REF#refs/heads/}
+          set +e
+          IS_BETA=$(echo ${current_branch} | grep '^v[0-9]\.[0-9]\.[0-9]-beta.*$')
+          set -e
+          if [ ! -z $IS_BETA ]
+          then
+            checkout_branch=beta
+          else
+            checkout_branch=main
+          fi
```

WDYT about using [this regexp](https://github.com/mesosphere/dkp-insights/blob/main/.github/workflows/release.yml#L77-L85) we used in our release builds in `dkp-insights` repo, its a better and more "official" one that the one I wrote here earlier in the TC job.
```suggestion
          # Regexp taken from SemVer 2.0 https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
          # Capture groups \4 (prerelease) and \5 (buildmetadata) are extracted here.
          PRERELEASE_BUILDMETADATA=$(echo ${{ inputs.majorVersion }}.${{ inputs.minorVersion }}.${{ inputs.patchVersion }} | perl -ne 'print "$4$5" if /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/')          
          if [ ! -z ${PRERELEASE_BUILDMETADATA} ]
          then
            checkout_branch=beta
          else
            checkout_branch=main
          fi
```

---

yes sure. I assume that we'll still use the `${current_branch}` instead of the inputs, as there's no inputs on this GHA workflow